app/src/main/java/com/yandex/practicum/middle_homework_4/ui/AppViewModel.kt

```diff
@@ -28,9 +27,19 @@ class AppViewModel(
 
     @OptIn(ExperimentalPagingApi::class)
     fun getNews(): Flow<PagingData<News>> =
-    // Допишите реализацию метода, используя класс Pager()
-    // Для реализации фабрики используйте newsDatabase
-    // Реализуйте NewsRemoteMediator() используя newsService и newsDatabase
+        Pager(
+            config = PagingConfig(
+                pageSize = PAGE_SIZE,
+                enablePlaceholders = true,
+                prefetchDistance = PREFETCH_DISTANCE,
+                initialLoadSize = PAGE_SIZE
+            ),
+            remoteMediator = NewsRemoteMediator(
+                newsService = newsService,
+                newsDatabase = newsDatabase
+            )
+        )
+        { newsDatabase.getNewsDao().getNews() }.flow
```

Comment:
В данном случае лучше закешировать этот флоу на уровне вьюмодели, чтобы избежать лишних перезагрузок данных:
```suggestion
        { newsDatabase.getNewsDao().getNews() }.flow.cachedIn(viewModelScope)
```